core/src/env/engine/off_chain/impls.rs

```diff
@@ -153,11 +155,15 @@ impl TypedEnv for EnvInstance {
             .map_err(Into::into)
     }
 
-    fn gas_price<T: EnvTypes>(&mut self) -> Result<T::Balance> {
-        self.chain_spec
+    /// Emulates gas price calculation
+    fn gas_price<T: EnvTypes>(&mut self, gas: u64) -> Result<T::Balance> {
```

Comment:
Would be nice to have a test for this function somewhere, since the logic now got more complex and logic surrounding cost analysis should be considered criticial imho. Besides this the code looks sound to me.

Reply:
Is it worth adding tests for the offchain environment?

Reply:
Theoretically yes they are worth. However, the off-chain environment currently lacks tests all over the place because originally it was not intended to stay for long since at the time of writing some standalone Seal implementation was still deemed to be realistic. Consider it a hack that came to stay.

Reply:
Added a test